pup-0003.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Cherry-Picking Commits
+author: David Davis
+created: 01-Apr-2017
+status:
+---
+
+## Summary
+
+Currently in Pulp, we merge features to master. However, hotfixes and bug fixes
+are merged to release branches which then are merged to newer release branches
+and then to master (a process called "merging forward"). More info about
+merging forward can be found in our documentation:
+
+http://docs.pulpproject.org/en/nightly/dev-guide/contributing/merging.html#merging-your-changes
+
+Also, a good illustration of this process is outlined in the following webpage:
+
+http://nvie.com/posts/a-successful-git-branching-model/
+
+This PUP is to use cherry-picking instead of merging forward. Using this model,
+all PRs would be made against master and then changes would be cherry-picked to
+release branches as needed.
```

Comment:
I wonder if we need to elaborate on the "as needed" part. As noted in one of my comments below, this approach could temp us to put fewer fixes onto the current release branch.

Right now all bug fixes default to landing on the current release branch and master; does this PUP propose changing that default? Either way, I think it's worth stating explicitly.